drizzlepac/runsinglehap.py

```diff
@@ -53,13 +53,29 @@ def perform(input_filename, **kwargs):
     debug : Boolean
         display all tracebacks, and debug information?
 
+    log_level : string
+        The desired level of verboseness in the log statements displayed on the screen and written to the .log file.
+
     Updates
     -------
     return_value : list
         a simple status value. '0' for a successful run and '1' for a failed
         run
     """
-    log.info("Starting single-visit processing of {}".format(input_filename))
+    # set up log_level as an input to hapsequencer.run_hap_processing().
+    if kwargs['log_level'] == "critical":
+        kwargs['log_level'] = logutil.logging.CRITICAL
+    elif kwargs['log_level'] == "error":
+        kwargs['log_level'] = logutil.logging.ERROR
+    elif kwargs['log_level'] == "warning":
+        kwargs['log_level'] = logutil.logging.WARNING
+    elif kwargs['log_level'] == "info":
+        kwargs['log_level'] = logutil.logging.INFO
+    elif kwargs['log_level'] == "debug":
+        kwargs['log_level'] = logutil.logging.DEBUG
+    else:
+        kwargs['log_level'] = logutil.logging.INFO
+
```

Comment:
The logic could be simplified to only 1 if_else statement (no elif's) with the use of a dict which translates between user-input values and logging.[CRITICAL|ERROR|...] values.  Also, user-input should be converted to all lower case for the comparison to avoid the code missing what the user 'intended' in case they try to replicate logging level names (all upper-case).

Reply:
Resolved by commit [623790e](https://github.com/spacetelescope/drizzlepac/pull/506/commits/623790ee0698e3d9b632399993f5f30c5f3266a6)